internal/handlers/pg/msg_find.go

```diff
@@ -128,7 +148,7 @@ func (h *Handler) fetchAndFilterDocs(ctx context.Context, tx pgx.Tx, sqlParam *p
 		_, doc, err := iter.Next()
 		if err != nil {
 			if errors.Is(err, iterator.ErrIteratorDone) {
-				return resDocs, nil
+				break
```

Comment:
Here I wanted to make it a bit easier to find where the result would be returned.

Reply:
You still need to read the whole function to check if there are other returns. But with `break`, you also need to read it backward from this point to check how many `for` there are.

`return` there is an extension of a general rule: return early. It is useful not only for error handling.